org.lflang/src/org/lflang/generator/ts/TSReactorGenerator.kt

```diff
@@ -0,0 +1,715 @@
+            indent(reactorConstructor)
+            pr(reactorConstructor, arguments.joinToString(", \n"))
+            unindent(reactorConstructor)
+            pr(reactorConstructor, ") {")
+            indent(reactorConstructor)
+        } else {
+            pr(reactorConstructor, "constructor (")
+            indent(reactorConstructor)
+            pr(reactorConstructor, arguments.joinToString(", \n"))
+            unindent(reactorConstructor)
+            pr(reactorConstructor, ") {")
+            indent(reactorConstructor)
+        }
+
+        var superCall: String
+        if (reactor.isMain()) {
+            superCall = "super(timeout, keepAlive, fast, success, fail);"
+        } else if (reactor.isFederated()) {
+            var port = federationRTIProperties()["port"]
+            // Default of 0 is an indicator to use the default port, 15045.
+            if (port === 0) {
+                port = 15045
+            }
+            superCall = """
+            super(${federate.id}, ${port},
+                "${federationRTIProperties()["host"]}",
+                timeout, keepAlive, fast, success, fail);
+            """
+        } else {
+            superCall = "super(parent);"
+        }
+        pr(reactorConstructor, superCall)
+
+        // Next handle child reactors instantiations.
+        // If the app isn't federated, instantiate all
+        // the child reactors. If the app is federated
+        var childReactors: List<Instantiation>
+        if (!reactor.isFederated()) {
+            childReactors = reactor.instantiations
+        } else {
+            childReactors = LinkedList<Instantiation>()
+            childReactors.add(federate.instantiation)
+        }
+
+        for (childReactor in childReactors) {
+            pr(childReactor.getName() + ": " + childReactor.reactorClass.name +
+                    if (childReactor.typeParms.isEmpty()) {""} else {
+                        childReactor.typeParms.joinToString(", ", "<", ">") { it.toText() }})
+
+            var childReactorArguments = StringJoiner(", ");
+            childReactorArguments.add("this")
+
+            // Iterate through parameters in the order they appear in the
+            // reactor class, find the matching parameter assignments in
+            // the reactor instance, and write the corresponding parameter
+            // value as an argument for the TypeScript constructor
+            for (parameter in childReactor.reactorClass.toDefinition().parameters) {
+                childReactorArguments.add(getTargetInitializer(parameter, childReactor))
+            }
+
+            pr(reactorConstructor, "this." + childReactor.getName()
+                    + " = new " + childReactor.reactorClass.name +
+                    "(" + childReactorArguments + ")" )
+        }
+
+        // Next handle timers.
+        for (timer in reactor.timers) {
+            var timerPeriod: String
+            if (timer.period === null) {
```

Comment:
This is a weird way to put it. You don't need a `var`. You could initialize the variable like so:
```kotlin
            val timerPeriod: String =
                if (timer.period == null) "0" 
                else getTargetValue(timer.period)
```
or equivalently
```kotlin
                val timerPeriod = timer.period?.let { getTargetValue(it) } ?: "0"
```

Same for the other vars.